xonsh/built_ins.py

```diff
@@ -523,6 +523,13 @@ def run_subproc(cmds, captured=True):
             stderr = streams['stderr']
         uninew = (ix == last_cmd) and (not captured)
         alias = builtins.aliases.get(cmd[0], None)
+        if (alias is None
+            and builtins.__xonsh_env__.get('AUTO_CD')
+            and os.path.isdir(cmd[0])
+            and locate_binary(cmd[0], cwd=None) is None
+            and len(cmds)==1):
+            cmd.insert(0, 'cd')
+            return run_subproc(cmds, captured=False)
```

Comment:
Does this need to be recursive? I think you can just set `alias = builtins.aliases.get('cd', None)` here instead of returning.